File: com.bosch.fsp.hackathon.gen/src/com/bosch/nevonex/common/impl/AbsolutePosition.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.common.impl;

import com.bosch.nevonex.common.IAbsolutePosition;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.EObject;
import org.eclipse.emf.ecore.InternalEObject;

import org.eclipse.emf.ecore.impl.EObjectImpl;

/**
 * @generated
 */
public class AbsolutePosition extends EObjectImpl implements IAbsolutePosition {
	protected double altitude = 0.0;
	protected double latitude = 0.0;
	protected double longitude = 0.0;

	protected AbsolutePosition() {
		super();
	}

	@Override
	protected EClass eStaticClass() {
		return CommonPackage.Literals.ABSOLUTE_POSITION;
	}

	public double getAltitude() {
		return altitude;
	}

	public void setAltitude(double newAltitude) {
		altitude = newAltitude;
	}

	public double getLatitude() {
		return latitude;
	}

	public void setLatitude(double newLatitude) {
		latitude = newLatitude;
	}

	public double getLongitude() {
		return longitude;
	}

	public void setLongitude(double newLongitude) {
		longitude = newLongitude;
	}

	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
		case CommonPackage.ABSOLUTE_POSITION__ALTITUDE:
			return getAltitude();
		case CommonPackage.ABSOLUTE_POSITION__LATITUDE:
			return getLatitude();
		case CommonPackage.ABSOLUTE_POSITION__LONGITUDE:
			return getLongitude();
		}
		return super.eGet(featureID, resolve, coreType);
	}

	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
		case CommonPackage.ABSOLUTE_POSITION__ALTITUDE:
			setAltitude((Double) newValue);
			return;
		case CommonPackage.ABSOLUTE_POSITION__LATITUDE:
			setLatitude((Double) newValue);
			return;
		case CommonPackage.ABSOLUTE_POSITION__LONGITUDE:
			setLongitude((Double) newValue);
			return;
		}
		super.eSet(featureID, newValue);
	}

	@Override
	public void eUnset(int featureID) {
		switch (featureID) {
		case CommonPackage.ABSOLUTE_POSITION__ALTITUDE:
			setAltitude(0.0);
			return;
		case CommonPackage.ABSOLUTE_POSITION__LATITUDE:
			setLatitude(0.0);
			return;
		case CommonPackage.ABSOLUTE_POSITION__LONGITUDE:
			setLongitude(0.0);
			return;
		}
		super.eUnset(featureID);
	}

	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
		case CommonPackage.ABSOLUTE_POSITION__ALTITUDE:
			return altitude != 0.0;
		case CommonPackage.ABSOLUTE_POSITION__LATITUDE:
			return latitude != 0.0;
		case CommonPackage.ABSOLUTE_POSITION__LONGITUDE:
			return longitude != 0.0;
		}
		return super.eIsSet(featureID);
	}
} //AbsolutePosition

File: hackathon_CPP_SDK/src-gen/nevonex/common/AbsolutePosition.cpp
/*
 * nevonex/common/AbsolutePosition.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "AbsolutePosition.hpp"
#include <ecore/EObject.hpp>
#include <ecore/EClass.hpp>
#include "nevonex/common/CommonPackage.hpp"
#include <ecorecpp/mapping.hpp>
#include <algorithm>
#include <nevonex/fcb/SubscriberEnum.hpp>
#include <nevonex-fcal-platform/log/Logger.hpp>

#if BOOST_VERSION >= 106501
#define BOOST_STACKTRACE_HEADERS_FOUND
#include <boost/stacktrace.hpp>
#endif

/*PROTECTED REGION ID(AbsolutePosition.cpp) START*/
// Please, enable the protected region if you add manually written code.
// To do this, add the keyword ENABLED before START.
/*PROTECTED REGION END*/

using namespace ::nevonex::common;
using namespace ::nevonex::log;

// Default constructor
AbsolutePosition::AbsolutePosition() : m_altitude(0.0), m_latitude(0.0), m_longitude(0.0)
{

    /*PROTECTED REGION ID(AbsolutePosition__AbsolutePosition) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

AbsolutePosition::~AbsolutePosition()
{

    /*PROTECTED REGION ID(AbsolutePosition__Destructor) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

// Attributes

::ecore::EDouble AbsolutePosition::getAltitude() const
{
    return m_altitude;
}

void AbsolutePosition::setAltitude(::ecore::EDouble _altitude)
{

    ::ecore::EDouble _old_altitude = m_altitude;
    m_altitude = _altitude;

}

::ecore::EDouble AbsolutePosition::getLatitude() const
{
    return m_latitude;
}

void AbsolutePosition::setLatitude(::ecore::EDouble _latitude)
{

    ::ecore::EDouble _old_latitude = m_latitude;
    m_latitude = _latitude;

}

::ecore::EDouble AbsolutePosition::getLongitude() const
{
    return m_longitude;
}

void AbsolutePosition::setLongitude(::ecore::EDouble _longitude)
{

    ::ecore::EDouble _old_longitude = m_longitude;
    m_longitude = _longitude;

}

// References

